repository/release.go

```diff
@@ -2,36 +2,57 @@ package repository
 
 import (
 	"context"
+	"errors"
 
+	"release-manager/pkg/apierrors"
 	"release-manager/repository/model"
-	"release-manager/repository/util"
+	"release-manager/repository/query"
 	svcmodel "release-manager/service/model"
 
-	"github.com/nedpals/supabase-go"
+	"github.com/georgysavva/scany/v2/pgxscan"
+	"github.com/google/uuid"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
 )
 
-const releaseDBEntity = "releases"
-
 type ReleaseRepository struct {
-	client *supabase.Client
+	dbpool *pgxpool.Pool
 }
 
-func NewReleaseRepository(c *supabase.Client) *ReleaseRepository {
+func NewReleaseRepository(pool *pgxpool.Pool) *ReleaseRepository {
 	return &ReleaseRepository{
-		client: c,
+		dbpool: pool,
 	}
 }
 
 func (r *ReleaseRepository) Create(ctx context.Context, rls svcmodel.Release) error {
-	data := model.ToRelease(rls)
-
-	err := r.client.
-		DB.From(releaseDBEntity).
-		Insert(&data).
-		ExecuteWithContext(ctx, nil)
+	_, err := r.dbpool.Exec(ctx, query.CreateRelease, pgx.NamedArgs{
+		"id":           rls.ID,
+		"projectID":    rls.ProjectID,
+		"releaseTitle": rls.ReleaseTitle,
+		"releaseNotes": rls.ReleaseNotes,
+		"createdBy":    rls.AuthorUserID,
+		"createdAt":    rls.CreatedAt,
+		"updatedAt":    rls.UpdatedAt,
+	})
 	if err != nil {
-		return util.ToDBError(err)
+		return err
 	}
 
 	return nil
 }
+
+func (r *ReleaseRepository) Read(ctx context.Context, id uuid.UUID) (svcmodel.Release, error) {
+	var rls model.Release
+
+	err := pgxscan.Get(ctx, r.dbpool, &rls, query.ReadRelease, pgx.NamedArgs{"id": id})
+	if err != nil {
+		if errors.Is(err, pgx.ErrNoRows) {
+			return svcmodel.Release{}, apierrors.NewReleaseNotFoundError().Wrap(err)
```

Comment:
Repository returns apierror directly

Reply:
The new approach is that all packages that "serve" the service will now return a service error. 

Currently, the errors are located in `pkg/apierrors`, but once the refactoring is complete, these errors will be moved to the `service` package, and `responseerrors` will be relocated to the transport package.